ast_tools/pattern.py

```diff
@@ -0,0 +1,109 @@
+import re
+import ast
+import itertools
+
+# Globals dictionary to evaluate type names in node patterns
+eval_globals = {}
+exec('from ast import *', eval_globals)
+
+
+class NodePattern:
+    """An individual node match, e.g. {x:Name}"""
+    def __init__(self, s):
+        self._parse(s)
+
+    def _parse(self, s):
+        """
+        Extracts the name and type of the desired AST node.
+        If no type is provided, then self.type is None.
+
+        Type names are evaluated in a context with the ast module imported.
+        """
+        parts = tuple(s[1:-1].split(':'))
+        if len(parts) == 1:
+            self.name = parts[0]
+            self.type = None
+        else:
+            assert (len(parts) == 2)
+            self.name = parts[0]
+            self.type = eval(parts[1], eval_globals)
```

Comment:
Again, this should work fine, but @cdonovick would probably say to just fetch the type from the `ast` module by name.  Particularly because this is just using a `str` name lookup right? (If the user could put in arbitrary code, e.g. a function that returns a type, then `eval` would probably be necessary)